code/go/0chain.net/smartcontract/storagesc/challenge.go

```diff
@@ -407,6 +474,64 @@ func (sc *StorageSmartContract) verifyChallenge(t *transaction.Transaction,
 	// verification, or partial verification
 	if pass && fresh {
 
+		blobber, err := sc.getBlobber(t.ClientID, balances)
+		if err != nil {
+			return "", common.NewError("verify_challenge",
+				"can't get blobber"+err.Error())
+		}
+
+		// this expiry of blobber needs to be corrected once logic is finalized
+
+		if blobber.RewardPartition.StartRound != rewardRound ||
+			balances.GetBlock().Round == 0 {
+
+			var dataRead float64 = 0
+			if blobber.LastRewardDataReadRound >= rewardRound {
+				dataRead = blobber.DataReadLastRewardRound
```

Comment:
Hmm, in what situation the blobber.ReadPatition.StartRound != rewardRound but the blobber.LastRewardDataReadRound  >= rewardRound could happen?

Perhaps we need to make sure the blobber.ReadPartition.StartRound and the LastRewardDataReadRound are updated correctly.

Reply:
when in this reward round period, data has been read by blobber, but verify_challenge has not been called yet for that blobber.

which means- in current reward round period, data has been accessed, but blobber has not yet been added to partition.